Optimiser.fs

```diff
@@ -39,6 +39,25 @@ module Types =
     open Starling.Core.Graph
 
     /// <summary>
+    ///     A type of unification used with the <c>Unify</c> leg of
+    ///     <see cref="Transform"/>.
+    /// <summary>
+    type UnifyMode =
```

Comment:
I've paged this out a bit, but in some cases where we're merging two nodes, we want to back out if there are some connections between them, and in some cases we just want to just turn those connections into cycles or delete them.

This could be done by adding a load of side-conditions and additional operations onto unification, which might be a bit less extravagant, but presumably 1-month-ago me thought that was too messy.